src/billing/Workspaces/Workspaces.tsx

```diff
@@ -222,58 +269,157 @@ interface WorkspacesProps {
-    field: 'name',
-    direction: 'asc',
+    field: isFeaturePreviewEnabled(SPEND_REPORTING) ? 'totalSpend' : 'name',
+    direction: isFeaturePreviewEnabled(SPEND_REPORTING) ? 'desc' : 'asc',
   });
   const [expandedWorkspaceName, setExpandedWorkspaceName] = useState<string>();
 
   const getBillingAccountStatus = (workspace: WorkspaceInfo): BillingAccountStatus =>
     // @ts-ignore
     _.findKey((g) => g.has(workspace), groups);
 
-  return _.isEmpty(workspacesInProject) ? (
-    <div style={{ ...Style.cardList.longCardShadowless, width: 'fit-content' }}>
-      <span aria-hidden='true'>Use this Terra billing project to create</span>
-      <Link
-        aria-label='Use this Terra billing project to create workspaces'
-        style={{ marginLeft: '0.3em', textDecoration: 'underline' }}
-        href={Nav.getLink('workspaces')}
-      >
-        Workspaces
-      </Link>
-    </div>
-  ) : (
-    !_.isEmpty(workspacesInProject) && (
-      <div role='table' aria-label={`workspaces in billing project ${billingProject.projectName}`}>
-        <WorkspaceCardHeaders
-          needsStatusColumn={billingAccountsOutOfDate}
-          sort={workspaceSort}
-          onSort={setWorkspaceSort}
-        />
-        <div>
-          {_.flow(
-            _.orderBy([workspaceSort.field], [workspaceSort.direction]),
-            _.map((workspace: WorkspaceInfo) => {
-              const isExpanded = expandedWorkspaceName === workspace.name;
-              return (
-                <WorkspaceCard
-                  workspace={workspace}
-                  billingAccountDisplayName={
-                    isGoogleWorkspaceInfo(workspace)
-                      ? billingAccounts[workspace.billingAccount]?.displayName
-                      : undefined
-                  }
-                  billingProject={billingProject}
-                  billingAccountStatus={billingAccountsOutOfDate && getBillingAccountStatus(workspace)}
-                  key={workspace.workspaceId}
-                  isExpanded={isExpanded}
-                  onExpand={() => setExpandedWorkspaceName(isExpanded ? undefined : workspace.name)}
-                />
-              );
-            })
-          )(workspacesInProject)}
+  const signal = useCancellation();
+
+  const [updating, setUpdating] = useState(false);
+  const [selectedDays, setSelectedDays] = useState<number>(30);
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [allWorkspacesInProject, setAllWorkspacesInProject] = useState<WorkspaceInfo[]>(workspacesInProject);
+
+  useEffect(() => {
+    const getSpendReportByWorkspaceAndCategory = (selectedDays: number, workspacesInProject: WorkspaceInfo[]) => {
+      const startDate = subDays(selectedDays, new Date()).toISOString().slice(0, 10);
+      const endDate = new Date().toISOString().slice(0, 10);
+      const aggregationKeys = ['Workspace~Category'];
+
+      setUpdating(true);
+      Ajax(signal)
```

Comment:
We shouldn't issue this call if there aren't any workspaces in the billing project